cpp/tests/io/csv_test.cpp

```diff
@@ -1169,4 +1170,46 @@ TEST_F(CsvReaderTest, UserImplementedSource)
   expect_column_data_equal(int32_values, view.column(2));
 }
 
+TEST_F(CsvReaderTest, DurationsWithWriter)
+{
+  auto filepath = temp_env->get_temp_dir() + "DurationsWithWriter.csv";
+
+  constexpr long max_value_d  = std::numeric_limits<cudf::duration_D::rep>::max();
+  constexpr long min_value_d  = std::numeric_limits<cudf::duration_D::rep>::min();
+  constexpr long max_value_ns = std::numeric_limits<cudf::duration_s::rep>::max();
+  constexpr long min_value_ns = std::numeric_limits<cudf::duration_s::rep>::min();
+  column_wrapper<cudf::duration_D, cudf::duration_D::rep> durations_D{
+    {-86400L, -3600L, -2L, -1L, 0L, 1L, 2L, min_value_d, max_value_d}};
+  column_wrapper<cudf::duration_s, int64_t> durations_s{{-86400L,
+                                                         -3600L,
+                                                         -2L,
+                                                         -1L,
+                                                         0L,
+                                                         1L,
+                                                         2L,
+                                                         min_value_ns / 1000000000 + 1,
+                                                         max_value_ns / 1000000000}};
+  column_wrapper<cudf::duration_ms, int64_t> durations_ms{
+    {-86400L, -3600L, -2L, -1L, 0L, 1L, 2L, min_value_ns / 1000000 + 1, max_value_ns / 1000000}};
+  column_wrapper<cudf::duration_us, int64_t> durations_us{
+    {-86400L, -3600L, -2L, -1L, 0L, 1L, 2L, min_value_ns / 1000 + 1, max_value_ns / 1000}};
+  column_wrapper<cudf::duration_ns, int64_t> durations_ns{
+    {-86400L, -3600L, -2L, -1L, 0L, 1L, 2L, min_value_ns, max_value_ns}};
+
+  cudf::table_view input_table(std::vector<cudf::column_view>{
+    durations_D, durations_s, durations_ms, durations_us, durations_ns});
+  std::vector<std::string> names{"D", "s", "ms", "us", "ns"};
+
+  write_csv_helper(filepath, input_table, true, names);
+
+  cudf_io::read_csv_args in_args{cudf_io::source_info{filepath}};
+  in_args.names = names;
+  in_args.dtype = {
+    "timedelta[D]", "timedelta64[s]", "timedelta64[ms]", "timedelta64[us]", "timedelta64[ns]"};
+  auto result = cudf_io::read_csv(in_args);
+
+  const auto result_table = result.tbl->view();
+  CUDF_TEST_EXPECT_TABLES_EQUIVALENT(input_table, result_table);
```

Comment:
This tests the isomorphism of the table with the written file, but it technically does not test that the written file contains the correctly formatted table. Should we have another test which verifies the contents of the file?

Reply:
I had same doubt as well. None of these tests verifies the contents of the file.

Reply:
Opened https://github.com/rapidsai/cudf/issues/6222 so we can track this. Might have similar suspect tests in other parts of the code.